components/service/glean/docs/internal.md

```diff
@@ -0,0 +1,27 @@
+# Internal behavior
+
+## Clearing metrics when disabling/enabling Glean
+
+When disabling upload (`Glean.setUploadEnabled(false)`), metrics are also
+cleared to prevent their storage on the local device, and lessen the likelihood
+of accidentally sending them.  There are a couple of exceptions to this:
+
+- `first_run_date` is retained so it isn't reset if metrics are later re-enabled.
+
+- `client_id` is set to the special value
+  `"c0ffeec0-ffee-c0ff-eec0-ffeec0ffeec0"`. This should make it possible to
+  detect the error case when metrics are sent from a client that has been disabled.
+  
+When re-enabling metrics:
+
+- `first_run_date` is left as-is. (It should remain a correct date of first run
+  of the application, unaffected by disabling/enabling Glean).
+
+- `client_id` is regenerated. It has no connection to the `client_id` used prior
```

Comment:
Nit: "`client_id` is set to a new UUID" or so? I.e. call out clearly that it gets a new UUID value, "regenerated" is not very clear to me on whats happening.

Reply:
Good point.  Updated the text here.